sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -600,7 +599,20 @@ public void setValueDeserializer(String valueDeserializer) {
      */
     public Read<K, V> withKeyDeserializerAndCoder(
         Class<? extends Deserializer<K>> keyDeserializer, Coder<K> keyCoder) {
-      return toBuilder().setKeyDeserializer(keyDeserializer).setKeyCoder(keyCoder).build();
+      return withKeyDeserializer(keyDeserializer).toBuilder().setKeyCoder(keyCoder).build();
+    }
+
+    /**
+     * Specify Confluent schema registry url and subject to get a deserializer (and schema) from
+     * Confluent Schema Registry.
+     */
+    public Read<K, V> withKeyDeserializer(String schemaRegistryUrl, String subject) {
```

Comment:
I'd suggest to extract these two string arguments into dedicated Confluent Schema Registry input arguments object to avoid potential collisions with other `withKeyDeserializer(String, String)` methods that can be added in the future. Also, it can be extended (if needed) and it will make clear that this is used only for CSR's schema.

Reply:
Let's remove completely the method signature and let users use the `DeserializerProvider` verion that way they will put explicitly the Confluent variant with their arguments.